unblob/handlers/archive/_safe_tarfile.py

```diff
@@ -77,23 +77,33 @@ def extract(self, tarinfo: tarfile.TarInfo, extract_root: Path):  # noqa: C901
         # prevent traversal attempts through links
         if tarinfo.islnk() or tarinfo.issym():
             if Path(tarinfo.linkname).is_absolute():
-                self.record_problem(
-                    tarinfo,
-                    "Absolute path as link target.",
-                    "Converted to extraction relative path.",
-                )
 
                 def calculate_linkname():
                     root = extract_root.resolve()
                     path = (extract_root / tarinfo.name).resolve()
-                    common_path = Path(os.path.commonpath([root, path]))
-                    # normally root == common_path
-                    # if it is not, the output will be bad
-                    depth = max(0, len(path.parts) - len(common_path.parts) - 1)
+
+                    if path.parts[: len(root.parts)] != root.parts:
+                        return None
```

Comment:
This would be easier to understand I think:

```python
if common_path != root:
    return None
```

But maybe I'm missing something.

Reply:
yeah, except `common_path` is no more with the last change.